sympy/solvers/bivariate.py

```diff
@@ -149,17 +149,18 @@ def _lambert(eq, x):
     sol = []
     # check only real solutions:
     for k in [-1, 0]:
-        l = LambertW(d/(a*b)*exp(c*d/a/b)*exp(-f/a), k)
+        l = d/(a*b)*exp(c*d/a/b)*exp(-f/a)
         # if W's arg is between -1/e and 0 there is
         # a -1 branch real solution, too.
-        if k and not l.is_real:
-            continue
-        rhs = -c/b + (a/d)*l
+        rhs_1 = -c/b + (a/d)*LambertW(l, k)
+        rhs_2 = -c/b + (a/d)*LambertW(-l, k)
 
         solns = solve(X1 - u, x)
         for i, tmp in enumerate(solns):
-            solns[i] = tmp.subs(u, rhs)
-            sol.append(solns[i])
+            if tmp.subs(u, rhs_1).is_real is not False or k == 0:
```

Comment:
What if `.is_real` is False and `k == -1`, are there any examples that fall in this category?

Also, if `.is_real` is `None` the solution will be added. Is it expected? otherwise `if tmp.subs(u, rhs_1)` would be suffice.

Reply:
>What if .is_real is False and k == -1, are there any examples that fall in this category?

No there is not any example in which we are considering real solutions.
>Also, if .is_real is None the solution will be added. Is it expected? otherwise if tmp.subs(u, rhs_1) would be suffice.

Yes because examples containing other symbols gives `.is_real` None by which the solutions obtained in that case are checked by `checksol` so that correct solutions are returned.

Reply:
> No there is not any example in which we are considering real solutions.

Then probably we will need one just to ensure the coverage. If someone (by mistake) in future removes this then there should be a test that fails.

Reply:
> No there is not any example in which we are considering real solutions.

Then probably we will need one just to ensure the coverage. If someone (by mistake) in future removes this then there should be a test that fails.

Reply:
Is this test ok?
```
 assert solve(x**2 - y**2/exp(x), x, y, dict=True) != [{x: 2*LambertW(-y/2)}, {x: 2*LambertW(y/2)}, {x: 2*LambertW(-y/2, -1)}, {x: 2*LambertW(y/2, -1)}]
```
These are the solutions when we remove the condition of .is_real .